calculate.tf

```
@@ -0,0 +1,20 @@
+locals {
+  // If a user specifies '.' for dns-name,
+  //   we are going to use ${env}.${domain} as the fqdn instead of ${dns-name}.${env}.${domain}
+  dns_name_chunk = data.ns_subdomain.this.dns_name == "." ? "" : "${data.ns_subdomain.this.dns_name}."
+
+  // If user specifies var.create_vanity,
+  //   we are going to drop ${env} from the fqdn
+  env_chunk = var.create_vanity ? "" : "${data.ns_workspace.this.env_name}."
+
+  subdomain_chunk = "${local.dns_name_chunk}${local.env_chunk}"
+  subdomain       = trimsuffix(local.subdomain_chunk, ".")
+  fqdn            = "${local.subdomain_chunk}${local.domain_name}"
+
+  is_passthrough = local.fqdn == local.domain_name
+
+  // output locals
+  name        = !local.is_passthrough ? google_dns_managed_zone.this[0].name : local.domain_name
+  zone_id     = !local.is_passthrough ? data.ns_workspace.this.block_ref : local.domain_zone_id
```

```suggestion
  zone_id     = !local.is_passthrough ? google_dns_managed_zone.this[0].name : local.domain_zone_id
```

This makes it clear to another contributor where this is coming from.
It also ensures the value is correct if someone changes the name of the zone.